pkg/config/remote/client/client.go

```diff
@@ -103,6 +104,9 @@ var defaultOptions = Options{pollInterval: 5 * time.Second}
 // TokenFetcher defines the callback used to fetch a token
 type TokenFetcher func() (string, error)
 
+// TLSClientConfigFetcher defines the callback used to fetch the IPC TLS Client configuration
+type TLSClientConfigFetcher func() *tls.Config
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Why take a callback and not the value directly ? 
Looks like we also do that for the auth token itself so it's coherent, but I'd be curious to know if there is a technical reason

Reply:
Same reason as above, due to historical reasons the creation of authentication artifacts is done in parallel and their creation time is unpredictable, we delay the fetch as much as possible to increase the chances of having them initialized.
This should be reworked to achieve better consistency.